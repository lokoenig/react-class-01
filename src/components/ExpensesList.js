import React from "react";
import { connect } from "react-redux";
import getFilteredExpenses from "../selectors/expenses";
import ExpenseItem from "./ExpenseItem";
const ExpensesList = (props) => {
    let template = (<p>No expenses entered</p>);
    if (props.expenses.length !== 0) {
        template = (
            <div>
                <h2>List of Expenses</h2>
            {
                    props.expenses.map( (o, index) => 
                    <ExpenseItem 
                    key={index} 
                    {...o}
                    />
                )
            }
            </div>
        );
    }
    return template;
};


const mapStateToProps = (state) => {
    return {
        expenses: getFilteredExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpensesList);
