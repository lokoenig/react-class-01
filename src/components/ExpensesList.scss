.expense-list {
    --pad: 1rem;
    display: flex;
    flex-direction: column;
    --grid-cols: ;



    &__column-header {
        display: grid;
        grid-template-columns: 5fr 1fr;

        background-color: var(--color-content-back-alt);
        padding: var(--pad);


        &--name {
        }

        &--amount {
            text-align: right;
        }
    }

    &__list {
        
        .expense-single{
            padding: .25rem var(--pad) .25rem var(--pad);
            text-decoration: none;
            color:var(--text-on-light);
            display: grid;
            grid-template-columns: 4fr 1fr 1fr;
            outline: 2px solid transparent;

            margin-bottom: .25rem;

            &:nth-child(even){
                background-color: var(--color-content-back-alt);
            }
            &__amount {
                text-align: right;
            }

            &:hover, &:focus-visible {
                outline: 2px solid var(--color-nav-back);
                background-color: var(--color-nav-back-alt);
                transition: 
                background-color .3s ease-in,
                outline-color .3s ease-in; 

            }
        }
    }
}