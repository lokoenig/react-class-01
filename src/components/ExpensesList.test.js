import React from "react";
import { render, screen } from '@testing-library/react';


import ExpenseTestData from "../fixtures/expenses";
import { ExpensesList } from "./ExpensesList";
import { BrowserRouter } from "react-router-dom";

test('renders ExpensesList component - with expenses', () => {
    const { getByText, asFragment } =
        render(
            <BrowserRouter>
                <ExpensesList expenses={ExpenseTestData} />
            </BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
});

test('renders ExpensesList component - no expenses', () => {
    const { getByText, asFragment } =
        render(
            <BrowserRouter>
                <ExpensesList expenses={[]} />
            </BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
});