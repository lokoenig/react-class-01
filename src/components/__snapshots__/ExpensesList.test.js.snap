// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ExpensesList component - no expenses 1`] = `
<DocumentFragment>
  <p>
    No expenses in meet the search criteria.
  </p>
</DocumentFragment>
`;

exports[`renders ExpensesList component - with expenses 1`] = `
<DocumentFragment>
  <div>
    <h2>
      List of Expenses
    </h2>
    <div
      class="expense-list-header"
    >
      <span>
        Description
      </span>
      <span
        class="total-amount"
        title="Total Expenses Amount"
      >
        $18,036.32
      </span>
      <span>
        Date Posted
      </span>
    </div>
    <div
      class="expense-single"
    >
      <span
        class="expense-single__description"
      >
        booze
      </span>
      <span
        class="expense-single__amount"
      >
        $140.00
      </span>
      <span
        class="expense-singe__created"
      >
        5/1/1995
      </span>
      <button>
        Edit
      </button>
    </div>
    <div
      class="expense-single"
    >
      <span
        class="expense-single__description"
      >
        rent
      </span>
      <span
        class="expense-single__amount"
      >
        $1,200.00
      </span>
      <span
        class="expense-singe__created"
      >
        3/27/1999
      </span>
      <button>
        Edit
      </button>
    </div>
    <div
      class="expense-single"
    >
      <span
        class="expense-single__description"
      >
        Cat Supplies
      </span>
      <span
        class="expense-single__amount"
      >
        $33.33
      </span>
      <span
        class="expense-singe__created"
      >
        3/27/2002
      </span>
      <button>
        Edit
      </button>
    </div>
    <div
      class="expense-single"
    >
      <span
        class="expense-single__description"
      >
        combs
      </span>
      <span
        class="expense-single__amount"
      >
        $16,662.99
      </span>
      <span
        class="expense-singe__created"
      >
        1/20/2019
      </span>
      <button>
        Edit
      </button>
    </div>
  </div>
</DocumentFragment>
`;
